extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/LdapRecorder.java

```diff
@@ -47,7 +50,14 @@ public RuntimeValue<SecurityRealm> createRealm(LdapSecurityRealmRuntimeConfig ru
             ldapSecurityRealmBuilder.addDirectEvidenceVerification(false);
         }
 
-        return new RuntimeValue<>(ldapSecurityRealmBuilder.build());
+        SecurityRealm ldapRealm = ldapSecurityRealmBuilder.build();
+
+        if (ldapRealm instanceof CacheableSecurityRealm && runtimeConfig.cache().enabled()) {
```

Comment:
`ldapSecurityRealmBuilder` currently builds `LdapSecurityRealm` which `CacheableSecurityRealm`, but if that should ever change, we will ignore enabled cache and user will not know. That should be illegal state, or some warning etc.

Reply:
Fail point. I will add a warning log to the user that the configuration will be ignored.